husky_components/datagrid/main.js

```diff
@@ -1506,22 +1508,34 @@
                 if (this.options.selectedCounter) {
                     this.sandbox.dom.on(this.$el,
                         'click',
-                        function(event) {
-                            this.sandbox.dom.stopPropagation(event);
-                            if (!this.gridViews[this.viewId].showSelected) {
-                                return;
-                            }
-
-                            this.show = !this.show;
-                            this.gridViews[this.viewId].showSelected(this.show);
-
-                            this.updateSelectedCounter();
-                        }.bind(this),
+                        this.toggleSelectedClickHandler.bind(this),
                         '.selected-elements .show-selected'
                     );
                 }
             },
 
+            /**
+             * Handlers the click on the "show selected" and "show all" button.
+             *
+             * @param {Object} event The click event
+             */
+            toggleSelectedClickHandler: function(event) {
+                this.sandbox.dom.stopPropagation(event);
+                if (!this.gridViews[this.viewId].showSelected) {
+                    return;
+                }
+
+                this.show = !this.show;
+                this.gridViews[this.viewId].showSelected(this.show);
+                if (!!this.show && !!this.paginations[this.paginationId]) {
+                    this.paginations[this.paginationId].destroy();
+                } else {
+                    this.rerenderPagination();
+                }
```

Comment:
The destroying of the pagination was added because the datagrid showed weird behavior when trying to paginate in the mode where only the selected records are shown. Moreover the `infinite-scroll` pagination had a pretty critical issue in this case, because as soon as the `show only selected` button was clicked, it thought that the bottom of the page has been reached and started paginating.